src/directives/vRotate.ts

```diff
@@ -31,9 +31,8 @@ export const vRotate = {
     const quaternion = new Quaternion().setFromAxisAngle(new Vector3(x.value, y.value, z.value)
       .normalize(), radiansPerFrame)
 
-    const { onLoop } = useRenderLoop()
-
-    onLoop(() => {
+    const { onBeforeRender } = useLoop()
```

Comment:
Directives appear to be outside of the `provide` hierarchy, so I don't think this will work directly.

In `playground/src/components/DirectiveSubComponent.vue`, try this code:

```html
<script setup lang="ts">
import { vRotate } from '@tresjs/core'
</script>

<template>
  <TresMesh v-rotate>
    <TresBoxGeometry />
    <TresMeshNormalMaterial />
  </TresMesh>
</template>
```


Error:

```
Uncaught (in promise) Error: useTresContext must be used together with useTresContextProvider
    useTresContext index.ts:296
    useLoop index.ts:31
    mounted vRotate.ts:34
```

Reply:
@alvarosabu what do we do with this directives?

We can commented (remove from the exports) until we find a good way to handler them
Or
We can let then use `useRenderLoop`for now

Reply:
@JaimeTorrealba @andretchen0 seems like if the loop is inside of the `mounted` method of the directive is called too soon (vRotate). `v-always-look-at` seems to work because the loop is inside of the method `update`. I don't think directives are meant to work with composables (setup)

Looking at the search https://github.com/search?q=v-rotate++language:vue&type=code&p=1 "nobody" at least with code on Github is using it, 99% of the search results are from another libraries aside TresJS.

If we keep them with `useRenderLoop` maybe we should add a warning in the docs that they would not work with renderMode `on-demand` or `manual`. But I will not block this feature because of the directives.

Reply:
unmm, no directive works well inside setup, and composables. V-rotate is the newest one, and the directive are more to "test" things, I wouldn't upload them into GitHub only in local.

Said that, I definitely don't want to block this one and/or confuse the users. If I've to take a decision, comment then (the ones who use the old loop). We can think better later.